web/blog/mobilizing-zeek-logs/index.md

```diff
@@ -0,0 +1,232 @@
+
+### Tab-Separated Values (TSV)
+
+Zeek's custom tab-separated value (TSV) format is variant of CSV with additional
+metadata, similar to a data frame.
+
+Here's how you create TSV logs from a trace:
+
+```bash
+zeek -C -r trace.pcap [scripts]
+```
+
+:::info disable checksumming
+We add `-C` to disable checksumming, telling Zeek to ignore mismatches and use
+all packets in the trace. This is good practice to process all packets in a
+trace, as some capturing setups may perturb the checksums.
+:::
+
+And here's a snippet of the corresponding `conn.log`:
+
+```
+#separator \x09
+#set_separator	,
+#empty_field	(empty)
+#unset_field	-
+#path	conn
+#open	2019-06-07-14-30-44
+#fields	ts	uid	id.orig_h	id.orig_p	id.resp_h	id.resp_p	proto	service	duration	orig_bytes	resp_bytes	conn_state	local_orig	local_resp	missed_bytes	history	orig_pkts	orig_ip_bytes	resp_pkts	resp_ip_bytes	tunnel_parents	community_id
+#types	time	string	addr	port	addr	port	enum	string	interval	count	count	string	bool	bool	count	string	count	count	count	count	set[string]	string
+1258531221.486539	Cz8F3O3rmUNrd0OxS5	192.168.1.102	68	192.168.1.6
+7	udp	dhcp	0.163820	301	300	SF	-	-	0	Dd	1	329	1	328	-	1:aWZfLIquYlCxKGuJ62fQGlgFzAI=
+1258531680.237254	CeJFOE1CNssyQjfJo1	192.168.1.103	137	192.168.1.255	137	udp	dns	3.780125	350	0	S0	-	-	546	0	0	-	1:fLbpXGtS1VgDhqUW+WYaP0v+NuA=
+```
+
+Many Zeek users would now resort to their downstream log management tool,
+assuming it supports the custom TSV format. Zeek also comes with small helper
+utility `zeek-cut` for light-weight reshaping of this TSV format. For example:
+
+```bash
+zeek-cut id.orig_h id.resp_h < conn.log
+```
+
+This selects the columns `id.orig_h` and `id.resp_h`. Back in the days, many
+folks used `awk` to extract fields by their position, e.g., with `$4`, `$7`,
+`$9`. This is not only difficult to understand, but also brittle, since Zeek
+schemas can change based on configuration. With `zeek-cut`, it's at least a bit
+more robust.
+
+With Tenzir, it's as easy to process Zeek TSV logs using the native
+[`zeek-tsv`](/next/formats/zeek-tsv) parser:
+
+```bash
+cat *.log | tenzir 'read zeek-tsv | select id.orig_h, id.resp_h'
+```
+
+Tenzir takes care of parsing the type information properly and keeps IP
+addresses and timestamps as native data types. You can also see in the examples
+that Tenzir handles multiple concatenated TSV logs of different schemas as you'd
+expect.
+
+Now that Zeek logs are flowing, you can do a lot more than selecting specific
+columns. Check out the [reshaping guide](/next/user-guides/reshape-data) for
+filtering rows, performing aggregations, or routing them elsewhere.
+
+### JSON
+
+Zeek can also render logs as JSON by setting
+[`LogAscii::use_json=T`](https://docs.zeek.org/en/master/frameworks/logging.html):
+
+```bash
+zeek -r trace.pcap LogAscii::use_json=T
+```
+
+As with TSV, this generates one file per log type containing the NDJSON records.
+Here are the same three entries from above in NDJSON:
```

Comment:
This reads a bit strange, do we need to say "NDJSON" twice?
Also, it looks like we only have two events in the TSV.

Reply:
I removed one event and one mention of "NDJSON".